Automattic/wp-calypso
# Atomic Transfer State: remove persistence and other complexity

Inspired by the Redux persistence work in #50222 and #50625, this PR simplifies the Redux code for fetching Atomic Transfer status displayed by the "Checkout Thank You" page after one purchases the eCommerce plan.

First, we don't need persistence for this state. The transfer is a short process that takes around a minute and its status becomes uninteresting after its done. And after reloading the "Thank You" page, the status is immediately re-fetched without needing to remember anything in local storage.

Second, we don't need the `fetchingTransfer` boolean and the `isFetchingTransfer` selector. It's used only to avoid dispatching `fetchAtomicTransfer` in case it's already fetching. But the data layer will do that deduplication automatically. We can just start fetching on mount and not worry about it.

Third, there were several `ATOMIC_TRANSFER_*` actions used only by the `fetchingTransfer` reducer. These can be now removed. We only need `ATOMIC_TRANSFER_REQUEST` to trigger the request and `ATOMIC_TRANSFER_SET` to receive the result.

Last, there are a few janitorial changes in the affected code:
- ~remove export of the unconnected `CheckoutThankYou` because it's not used anywhere~
- ~remove default for the `fetchAtomicTransfer` because we know that `connect` will always supply it~
- remove few usages of `get` in favor of `?.` and `??`
- use a more concise way how to `mapDispatchToProps`

**How to test:**
Purchase an eCommerce plan with your credits. After payment, you'll be redirected to the `/checkout/thank-you/:site/:receipt` page that will show that transfer is in progress:

<img width="447" alt="Screenshot 2021-03-10 at 10 11 09" src="https://user-images.githubusercontent.com/664258/110610017-1d30ab00-818e-11eb-80ff-4584e282436d.png">

and shows an "Atomic Thank You" card after the transfer is complete:

<img width="510" alt="Screenshot 2021-03-10 at 10 20 01" src="https://user-images.githubusercontent.com/664258/110610101-2f124e00-818e-11eb-9072-5669b3a4dc99.png">

Verify that the "Thank You" UI shows correct status even after page reload.
